serverscommands/instancecommands/list.go

```diff
@@ -89,58 +91,49 @@ func tableList(c *cli.Context, i interface{}) {
 		fmt.Fprintf(c.App.Writer, "Could not type assert interface\n%+v\nto []osServers.Server\n", i)
 		os.Exit(1)
 	}
-	t := tablewriter.NewWriter(c.App.Writer)
-	t.SetAlignment(tablewriter.ALIGN_LEFT)
+
 	keys := []string{"ID", "Name", "Status", "Public IPv4", "Private IPv4", "Image", "Flavor"}
-	t.SetHeader(keys)
+
+	w := new(tabwriter.Writer)
+	w.Init(c.App.Writer, 0, 8, 0, '\t', 0)
+
+	// Write the header
+	fmt.Fprintln(w, strings.Join(keys, "\t"))
 	for _, server := range servers {
 		m := structs.Map(server)
-		f := []string{}
-		for _, key := range keys {
-			tmp := ""
-			switch key {
-			case "Public IPv4":
-				tmp = fmt.Sprint(m["AccessIPv4"])
-			case "Private IPv4":
-				i, ok := m["Addresses"].(map[string]interface{})
-				if !ok {
-					tmp = ""
-					break
-				}
-				j, ok := i["private"].([]interface{})
-				if !ok || len(j) == 0 {
-					tmp = ""
-					break
-				}
-				i, ok = j[0].(map[string]interface{})
-				if !ok {
-					tmp = ""
-					break
-				}
-				tmp = fmt.Sprint(i["addr"])
-			case "Image":
-				i, ok := m["Image"].(map[string]interface{})
-				if !ok {
-					tmp = ""
-					break
-				}
-				tmp = fmt.Sprint(i["id"])
-			case "Flavor":
-				i, ok := m["Flavor"].(map[string]interface{})
-				if !ok {
-					tmp = ""
-					break
+
+		// Extract the Image ID - if !ok, image is ""
+		image, _ := GetNestedID(m["Image"])
+
+		// Extract the Flavor ID - if !ok, flavor is ""
+		flavor, _ := GetNestedID(m["Flavor"])
+
+		// Extract the very first private address
+		// TODO: How do we handle multiples here?
+		privAddr := ""
+		a, ok := m["Addresses"].(map[string]interface{})
+		if ok {
+			a, ok := a["private"].([]interface{})
+			if ok && len(a) > 0 {
+				first, ok := a[0].(map[string]interface{})
+				if ok {
+					privAddr = fmt.Sprint(first["addr"])
 				}
-				tmp = fmt.Sprint(i["id"])
-			default:
-				tmp = fmt.Sprint(m[key])
-			}
-			if tmp == "<nil>" {
-				tmp = ""
+
 			}
-			f = append(f, tmp)
 		}
-		t.Append(f)
+
+		fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t%s\t%s\n", m["ID"], m["Name"], m["Status"], m["AccessIPv4"], privAddr, image, flavor)
+
+	}
+	w.Flush()
+}
+
+// GetNestedID extracts the nested id from, e.g. the flavor from a server
+func GetNestedID(i interface{}) (string, bool) {
```

Comment:
If this function can be used by commands in other packages, it should go in `util.go` (which will be refactored into smaller files). If it can be used by other subcommands in this package (`instancecommands`), it should be lowercase and put in `common.go`. If it can only be used by this command, it should still be lowercase.